core/event.mli

```diff
@@ -35,30 +35,25 @@ module Location : sig
 end
 
 module Ok : sig
-  module Trace : sig
+  module Data : sig
     type t =
-      { thread : Thread.t
-      ; time : Time_ns.Span.t
-      ; trace_state_change : Trace_state_change.t option
-      ; kind : Kind.t option
-      ; src : Location.t
-      ; dst : Location.t
-      }
-    [@@deriving sexp]
-  end
-
-  module Power : sig
-    type t =
-      { thread : Thread.t
-      ; time : Time_ns.Span.t
-      ; freq : int
-      }
+      | Trace of
+          { trace_state_change : Trace_state_change.t option
+          ; kind : Kind.t option
+          ; src : Location.t
+          ; dst : Location.t
+          } (** Represents an event collected from Intel PT. *)
+      | Sample of { callstack : Location.t list }
+          (** Represents event collected through sampling. *)
+      | Power of { freq : int } (** Power event collected by Intel PT. *)
```

Comment:
Nit: sort these grouped by sample type, i.e. have `Power` next to `Trace`.

Reply:
Could also nest by perf mode, along the lines of `Pt (Trace _) | Pt (Power _) | Sample _` since any given pipe will consist only of (`Trace` and `Power`) or `Sample`

Reply:
That does seem like a reasonable option. One thought though, is what if we supported power events which aren't part of Intel PT as well and used the same `Power` event? (it seems like you can get the same information through other perf events). And another thought, although we could group them under `Pt`, they aren't really being processed or handled together which makes me lean towards leaving them as is since it doesn't seem useful to pattern match only on `Pt` events.